dts/common/nordic/nrf54l20.dtsi

```diff
@@ -0,0 +1,614 @@
+				clock-frequency = <I2C_BITRATE_STANDARD>;
+				interrupts = <260 NRF_DEFAULT_IRQ_PRIORITY>;
+				easydma-maxcnt-bits = <16>;
+				status = "disabled";
+			};
+
+			spi30: spi@104000 {
+				/*
+				 * This spi node can be either SPIM or SPIS,
+				 * for the user to pick:
+				 * compatible = "nordic,nrf-spim" or
+				 *              "nordic,nrf-spis".
+				 */
+				compatible = "nordic,nrf-spim";
+				#address-cells = <1>;
+				#size-cells = <0>;
+				reg = <0x104000 0x1000>;
+				interrupts = <260 NRF_DEFAULT_IRQ_PRIORITY>;
+				max-frequency = <DT_FREQ_M(8)>;
+				easydma-maxcnt-bits = <16>;
+				rx-delay-supported;
+				rx-delay = <1>;
+				status = "disabled";
+			};
+
+			uart30: uart@104000 {
+				compatible = "nordic,nrf-uarte";
+				reg = <0x104000 0x1000>;
+				interrupts = <260 NRF_DEFAULT_IRQ_PRIORITY>;
+				status = "disabled";
+			};
+
+#ifdef USE_NON_SECURE_ADDRESS_MAP
+			/* intentionally empty because WDT30 is hardware fixed to Secure */
+#else
+			wdt30: watchdog@108000 {
+				compatible = "nordic,nrf-wdt";
+				reg = <0x108000 0x620>;
+				interrupts = <264 NRF_DEFAULT_IRQ_PRIORITY>;
+				status = "disabled";
+			};
+#endif
+
+			wdt31: watchdog@109000 {
+				compatible = "nordic,nrf-wdt";
+				reg = <0x109000 0x620>;
+				interrupts = <265 NRF_DEFAULT_IRQ_PRIORITY>;
+				status = "disabled";
+			};
+
+			gpio0: gpio@10a000 {
+				compatible = "nordic,nrf-gpio";
+				gpio-controller;
+				reg = <0x10a000 0x300>;
+				#gpio-cells = <2>;
+				ngpios = <5>;
+				status = "disabled";
+				port = <0>;
+				gpiote-instance = <&gpiote30>;
+			};
+
+			gpiote30: gpiote@10c000 {
+				compatible = "nordic,nrf-gpiote";
+				reg = <0x10c000 0x1000>;
+				status = "disabled";
+				instance = <30>;
+			};
+
+			clock: clock@10e000 {
+				compatible = "nordic,nrf-clock";
+				reg = <0x10e000 0x1000>;
+				interrupts = <270 NRF_DEFAULT_IRQ_PRIORITY>;
+				status = "disabled";
+			};
+		};
+
+		rram_controller: rram-controller@5004e000 {
+			compatible = "nordic,rram-controller";
+			reg = <0x5004e000 0x1000>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+			interrupts = <75 NRF_DEFAULT_IRQ_PRIORITY>;
+
+			cpuapp_rram: rram@0 {
+				compatible = "soc-nv-flash";
+				reg = <0x0 DT_SIZE_K(1428)>;
+				erase-block-size = <4096>;
+				write-block-size = <16>;
+			};
+			cpuflpr_rram: rram@165000 {
+				compatible = "soc-nv-flash";
+				reg = <0x165000 DT_SIZE_K(96)>;
+				erase-block-size = <4096>;
+				write-block-size = <16>;
+			};
+		};
+
+		cpuapp_ppb: cpuapp-ppb-bus {
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			cpuapp_nvic: interrupt-controller@e000e100  {
+				#address-cells = <1>;
+				compatible = "arm,v8m-nvic";
+				reg = <0xe000e100 0xc00>;
+				arm,num-irq-priority-bits = <3>;
+				interrupt-controller;
+				#interrupt-cells = <2>;
+			};
+
+			cpuapp_systick: timer@e000e010 {
+				compatible = "arm,armv8m-systick";
+				reg = <0xe000e010 0x10>;
+				status = "disabled";
+			};
+		};
+	};
```

Comment:
This section should be moved to a file like dts/arm/nordic/nrf54l20_cpuapp.dtsi. It should not be visible in FLPR builds.

Reply:
I would prefer to keep it and unify description for nRF devices later